src/commitment_scheme/kzg10/key.rs

```diff
@@ -141,14 +141,25 @@ impl CommitKey {
         Ok(truncated_powers)
     }
 
+    /// Checks whether the polynomial we are committing to:
+    /// - Has zero degree
+    /// - Has a degree which is more than the max supported degree
+    ///
+    /// Returns an error if any of the above conditions are true.
     fn check_commit_degree_is_within_bounds(&self, poly_degree: usize) -> Result<(), Error> {
-        check_degree_is_within_bounds(self.max_degree(), poly_degree)
+        if poly_degree == 0 {
```

Comment:
maybe this could be done better with a `match`, but it's up to you.

Reply:
Yep. Much better